### PR TITLE
update instead of replacing passwd file

### PR DESCRIPTION
**Pull request by bparees**

### CODE REVIEW

**Comment by bparees** [comment]
@mfojtik @mnagy ptal  (main concern is around the user 26 magic discussed in the Dockerfile)

File: 9.2/root/usr/share/container-scripts/postgresql/common.sh [hunk 1]
```diff
@@ -93,13 +93,22 @@ function generate_postgresql_recovery_config() {
...
-  export GROUP_ID=$(id -g)
-  grep -v ^postgres /etc/passwd > "$HOME/passwd"
-  echo "postgres:x:${USER_ID}:${GROUP_ID}:PostgreSQL Server:${HOME}:/bin/bash" >> "$HOME/passwd"
-  export LD_PRELOAD=libnss_wrapper.so
-  export NSS_WRAPPER_PASSWD=${HOME}/passwd
-  export NSS_WRAPPER_GROUP=/etc/group
+  USER_ID=$(id -u)
+  GROUP_ID=$(id -g)
+
+  if [ x"$USER_ID" != x"0" -a x"$USER_ID" != x"1001" ]; then
```
Comment (mnagy) [inline 9.2/root/usr/share/container-scripts/postgresql/common.sh]: I think the x's are unnecessary here.

File: 9.4/root/usr/share/container-scripts/postgresql/common.sh [hunk 2]
```diff
@@ -93,13 +93,22 @@ function generate_postgresql_recovery_config() {
...
-  export GROUP_ID=$(id -g)
-  grep -v ^postgres /etc/passwd > "$HOME/passwd"
-  echo "postgres:x:${USER_ID}:${GROUP_ID}:PostgreSQL Server:${HOME}:/bin/bash" >> "$HOME/passwd"
-  export LD_PRELOAD=libnss_wrapper.so
-  export NSS_WRAPPER_PASSWD=${HOME}/passwd
-  export NSS_WRAPPER_GROUP=/etc/group
+  USER_ID=$(id -u)
+  GROUP_ID=$(id -g)
+
+  if [ x"$USER_ID" != x"0" -a x"$USER_ID" != x"1001" ]; then
```
Comment (mnagy) [inline 9.4/root/usr/share/container-scripts/postgresql/common.sh]: same as above

File: 9.2/root/usr/share/container-scripts/postgresql/common.sh [hunk 3]
```diff
@@ -93,13 +93,22 @@ function generate_postgresql_recovery_config() {
...
+    NSS_WRAPPER_GROUP=/etc/group
+
+    grep -v ^postgres /etc/passwd > $NSS_WRAPPER_PASSWD
+    echo "postgres:x:${USER_ID}:${GROUP_ID}:PostgreSQL Server:${HOME}:/bin/bash" >> $NSS_WRAPPER_PASSWD
+
+    export NSS_WRAPPER_PASSWD
+    export NSS_WRAPPER_GROUP
+
+    LD_PRELOAD=libnss_wrapper.so
+    export LD_PRELOAD
```
Comment (mnagy) [inline 9.2/root/usr/share/container-scripts/postgresql/common.sh]: Why not one line?

**Comment by mnagy** [comment]
I'm a bit confused about the name of this PR. It seems like you are doing functionally the same thing wrt updating/replacing the passwd file, but you are using a different path.

And why the added userid check for 0 and 1001? Is that a bug fix for some problem?

File: 9.2/root/usr/share/container-scripts/postgresql/common.sh (hunk 1 above)
Comment (bparees) [inline 9.2/root/usr/share/container-scripts/postgresql/common.sh]: it's considered a good practice to ensure full compatibility:
http://stackoverflow.com/questions/174119/why-do-shell-script-comparisons-often-use-xvar-xyes

File: 9.2/root/usr/share/container-scripts/postgresql/common.sh (hunk 3 above)
Comment (bparees) [inline 9.2/root/usr/share/container-scripts/postgresql/common.sh]: symmetry w/ the others?  :) 

I think this was the result of some rework/reordering. i'll change it.

**Comment by bparees** [comment]
@mnagy it's addressing this issue:
https://github.com/openshift/s2i-python/issues/98

the point is to start from the image's /etc/passwd file instead of from a hardcoded template that may not contain all the same users.

the 0 and 1001 thing is copied from the way it was done in the s2i images to avoid setting up nss when we're running as the image user (which is 1001) because the image user exists in /etc/passwd already.  I think it can be removed for this image so i'll do that.

**Comment by bparees** [comment]
@mnagy hm.  ok now i'm not sure what i was doing here.  I think i was just blindly aligning this image w/ the other images i was updating and didn't stop to see this one was already doing the right thing.

will close this.

**Pull request closed without merging**